ports/stm32/omv_gpu.c

```diff
@@ -117,15 +117,23 @@ int omv_gpu_draw_image(image_t *src_img,
     nema_blit_subrect_fit(dst_rect->x, dst_rect->y, dst_rect->w, dst_rect->h,
                           src_rect->x, src_rect->y, src_rect->w, src_rect->h);
 
+    // Ensures any cached writes to dst are flushed.
+    SCB_CleanInvalidateDCache_by_Addr(dst_img->data, image_size(dst_img));
     // Flush source image
     SCB_CleanDCache_by_Addr(src_img->data, image_size(src_img));
+    // Ensure the GPU cache is clean before starting the GPU operation.
+    // Will start invalidating the GPU cache if not already invalidated.
+    HAL_ICACHE_Invalidate();
```

Comment:
Should this be `HAL_ICACHE_WaitForInvalidateComplete();` ?

Reply:
This would flush it twice after a draw. I think we should call `HAL_ICACHE_WaitForInvalidateComplete`

Reply:
Works, updated